caSlider improvements

dassjosh opened the pull request.

AntonMezger commented:
Dear Josh,

did I see correctly that you suppressed the possibility to have different directions for the slider (you just took the up slider) ?

could you comment on this ?


best regards

Anton
On 12 Apr 2016, at 19:59, Josh Dassinger <notifications@github.com<mailto:notifications@github.com>> wrote:

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/caqtdm/caqtdm/pull/14

Commit Summary

  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merged caqtdm development main into development fork
  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merge remote branch 'origin/Development' into Development
  *   Initial Argonne caSlider Enhancements
  *   Rebase fix attempt
  *   Attempt to merge from upstream
  *   caSlider: Moved setFocus to event. Removed overided enterEvent
  *   caSlider Fixes on QWT >= 6.1.x

File Changes

  *   M caQtDM_Lib/src/caqtdm_lib.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-0> (69)
  *   M caQtDM_Lib/src/inlines.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-1> (12)
  *   M caQtDM_Lib/src/limitsDialog.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-2> (141)
  *   M caQtDM_Lib/src/limitsDialog.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-3> (2)
  *   M caQtDM_QtControls/src/caslider.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-4> (348)
  *   M caQtDM_QtControls/src/caslider.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-5> (73)
  *   M caQtDM_Viewer/parser/QtProperties.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-6> (17)
  *   M caQtDM_Viewer/parser/parser.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-7> (9)

Patch Links:

  *   https://github.com/caqtdm/caqtdm/pull/14.patch
  *   https://github.com/caqtdm/caqtdm/pull/14.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14>

AntonMezger commented:
Dear Josh,

forget about that mail. I saw that the code just moved, sorry

best regards

Anton
On 12 Apr 2016, at 19:59, Josh Dassinger <notifications@github.com<mailto:notifications@github.com>> wrote:

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/caqtdm/caqtdm/pull/14

Commit Summary

  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merged caqtdm development main into development fork
  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merge remote branch 'origin/Development' into Development
  *   Initial Argonne caSlider Enhancements
  *   Rebase fix attempt
  *   Attempt to merge from upstream
  *   caSlider: Moved setFocus to event. Removed overided enterEvent
  *   caSlider Fixes on QWT >= 6.1.x

File Changes

  *   M caQtDM_Lib/src/caqtdm_lib.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-0> (69)
  *   M caQtDM_Lib/src/inlines.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-1> (12)
  *   M caQtDM_Lib/src/limitsDialog.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-2> (141)
  *   M caQtDM_Lib/src/limitsDialog.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-3> (2)
  *   M caQtDM_QtControls/src/caslider.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-4> (348)
  *   M caQtDM_QtControls/src/caslider.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-5> (73)
  *   M caQtDM_Viewer/parser/QtProperties.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-6> (17)
  *   M caQtDM_Viewer/parser/parser.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-7> (9)

Patch Links:

  *   https://github.com/caqtdm/caqtdm/pull/14.patch
  *   https://github.com/caqtdm/caqtdm/pull/14.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14>

AntonMezger commented:
Good job, I will take it over. However two small things:

in keypressevent you should call QwtAbstractSlider::setValue otherwise the keyboard up / down keys will not work
in set direction at the end the equal sign should be a different sign, i.e f(scalepos != NoScale) setScalePosition(scallops);

what is not quite nice, is the fact the the label for a vertical left scale goes over the slider. I did not find a solution yet

best regards

Anton


On 12 Apr 2016, at 19:59, Josh Dassinger <notifications@github.com<mailto:notifications@github.com>> wrote:

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/caqtdm/caqtdm/pull/14

Commit Summary

  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merged caqtdm development main into development fork
  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merge remote branch 'origin/Development' into Development
  *   Initial Argonne caSlider Enhancements
  *   Rebase fix attempt
  *   Attempt to merge from upstream
  *   caSlider: Moved setFocus to event. Removed overided enterEvent
  *   caSlider Fixes on QWT >= 6.1.x

File Changes

  *   M caQtDM_Lib/src/caqtdm_lib.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-0> (69)
  *   M caQtDM_Lib/src/inlines.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-1> (12)
  *   M caQtDM_Lib/src/limitsDialog.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-2> (141)
  *   M caQtDM_Lib/src/limitsDialog.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-3> (2)
  *   M caQtDM_QtControls/src/caslider.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-4> (348)
  *   M caQtDM_QtControls/src/caslider.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-5> (73)
  *   M caQtDM_Viewer/parser/QtProperties.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-6> (17)
  *   M caQtDM_Viewer/parser/parser.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-7> (9)

Patch Links:

  *   https://github.com/caqtdm/caqtdm/pull/14.patch
  *   https://github.com/caqtdm/caqtdm/pull/14.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14>

AntonMezger commented:
These changes have been integrated. However some changes had to be done, some defaults were changed for our lab and caThermo was also updated in order to be able to display a value

Enjoy
On 12 Apr 2016, at 19:59, Josh Dassinger <notifications@github.com<mailto:notifications@github.com>> wrote:

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/caqtdm/caqtdm/pull/14

Commit Summary

  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merged caqtdm development main into development fork
  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merge remote branch 'origin/Development' into Development
  *   Initial Argonne caSlider Enhancements
  *   Rebase fix attempt
  *   Attempt to merge from upstream
  *   caSlider: Moved setFocus to event. Removed overided enterEvent
  *   caSlider Fixes on QWT >= 6.1.x

File Changes

  *   M caQtDM_Lib/src/caqtdm_lib.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-0> (69)
  *   M caQtDM_Lib/src/inlines.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-1> (12)
  *   M caQtDM_Lib/src/limitsDialog.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-2> (141)
  *   M caQtDM_Lib/src/limitsDialog.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-3> (2)
  *   M caQtDM_QtControls/src/caslider.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-4> (348)
  *   M caQtDM_QtControls/src/caslider.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-5> (73)
  *   M caQtDM_Viewer/parser/QtProperties.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-6> (17)
  *   M caQtDM_Viewer/parser/parser.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-7> (9)

Patch Links:

  *   https://github.com/caqtdm/caqtdm/pull/14.patch
  *   https://github.com/caqtdm/caqtdm/pull/14.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14>

dassjosh commented:
Dear Anton,

I have incorporated added your suggestion in the keypressedevent.

In set direction the reason i have it as == is because none of the sliders get configured properly during the initial load when using QWT >= 6.1.0. I have attached two photos showing the difference between != and ==.
![caslider](https://cloud.githubusercontent.com/assets/14892282/14534573/66f760aa-022f-11e6-856a-d25abd3f796a.png)
![caslider](https://cloud.githubusercontent.com/assets/14892282/14534576/6a24754c-022f-11e6-82c2-ee44266098a6.png)

I do agree that it isn't nice when the label goes over the slider. I had a hard time with a solution since the font size of the label is tied to the font size of the scale.

Josh

AntonMezger commented:
I do not really understand what is happening there. It seems that I have no problems on my mac and linux plattform. Try to check out the new version and try again ? You may also send me your ui file.

best regards

Anton
On 12 Apr 2016, at 19:59, Josh Dassinger <notifications@github.com<mailto:notifications@github.com>> wrote:

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/caqtdm/caqtdm/pull/14

Commit Summary

  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merged caqtdm development main into development fork
  *   Initial Argonne caSlider Enhancements
  *   Initial Argonne caSlider Enhancements
  *   Merge remote branch 'origin/Development' into Development
  *   Initial Argonne caSlider Enhancements
  *   Rebase fix attempt
  *   Attempt to merge from upstream
  *   caSlider: Moved setFocus to event. Removed overided enterEvent
  *   caSlider Fixes on QWT >= 6.1.x

File Changes

  *   M caQtDM_Lib/src/caqtdm_lib.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-0> (69)
  *   M caQtDM_Lib/src/inlines.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-1> (12)
  *   M caQtDM_Lib/src/limitsDialog.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-2> (141)
  *   M caQtDM_Lib/src/limitsDialog.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-3> (2)
  *   M caQtDM_QtControls/src/caslider.cpp<https://github.com/caqtdm/caqtdm/pull/14/files#diff-4> (348)
  *   M caQtDM_QtControls/src/caslider.h<https://github.com/caqtdm/caqtdm/pull/14/files#diff-5> (73)
  *   M caQtDM_Viewer/parser/QtProperties.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-6> (17)
  *   M caQtDM_Viewer/parser/parser.c<https://github.com/caqtdm/caqtdm/pull/14/files#diff-7> (9)

Patch Links:

  *   https://github.com/caqtdm/caqtdm/pull/14.patch
  *   https://github.com/caqtdm/caqtdm/pull/14.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14>

dassjosh commented:
Here is the ui file that i am using. It made me add the .txt extension so remove that before use.
[sliderTest.ui.txt](https://github.com/caqtdm/caqtdm/files/219510/sliderTest.ui.txt)

I am currently working on merging your code with mine. 

Josh

dassjosh commented:
I just finished the merge and the displays look the same. 

Josh

AntonMezger commented:
The problem is that the file has been created with qwt6.0 where bottom scale was defined. However in version qwt6.1 these enums were replaced by leading and trailing scale. You will have to go through the designer again to fix that. I will see if I can patch that

best regards

Anton
On 14 Apr 2016, at 18:11, Josh Dassinger <notifications@github.com<mailto:notifications@github.com>> wrote:


Here is the ui file that i am using. It made me add the .txt extension so remove that before use.
sliderTest.ui.txt<https://github.com/caqtdm/caqtdm/files/219510/sliderTest.ui.txt>

I am currently working on merging your code with mine.

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-210025354>

AntonMezger commented:
I made a patch in caslider for files that where created for qwt6.0 conventions

hope that helps

On 14 Apr 2016, at 18:34, Josh Dassinger <notifications@github.com<mailto:notifications@github.com>> wrote:


I just finished the merge and the displays look the same.

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-210037860>

dassjosh commented:
I will pull your patch in. 
When i converted the file to use 6.1 I realized that the label for the vertical slider was on top of the actual slider. I think this was what you were saying before and I misinterpreted it. I have fixed that issue and it's now working as intended.

Josh

dassjosh commented:
Dear Anton,

I have pushed some small fixes up.

With the patch that you applied it fixed everything but the if(scalepos != NoScale) causes the scale to always appear when it should be no scale. So i removed the if.

I also found a break; that was missing when calculating where to place the vertical label so it always appeared on the right.

Josh

AntonMezger commented:
Dear Josh,

Thank a lot for your support and bug fixes. The two bug fixes have been integrated and in github

I missed these

From: Josh Dassinger [mailto:notifications@github.com]
Sent: Dienstag, 19. April 2016 17:37
To: caqtdm/caqtdm
Cc: Mezger Anton Christian (PSI)
Subject: Re: [caqtdm/caqtdm] caSlider improvements (#14)


Dear Anton,

I have pushed some small fixes up.

With the patch that you applied it fixed everything but the if(scalepos != NoScale) causes the scale to always appear when it should be no scale. So i removed the if.

I also found a break; that was missing when calculating where to place the vertical label so it always appeared on the right.

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-211983117>

dassjosh commented:
Dear Anton,

I have push some more bug fixes related to the slider.

1. If the sliders step was greater than distance between the slider and the limit it would allow a user to go outside the limits.
2. If the slider had a direction of Down or Left and one clicked or held the slider would move in the opposite direction of the click.
3. If the slider had a direction of Down or Left the Low limit and High limit was swapped allowing those sliders to go outside the limits.

Josh

AntonMezger commented:
Dear Josh,

Many thanks for your bug fixes ; we are not really using sliders, so that they are not really debugged. I do not know when I get these bug fixes in, while I am half an hour away from my holidays

Best regards

Anton

From: Josh Dassinger [mailto:notifications@github.com]
Sent: Donnerstag, 28. April 2016 17:36
To: caqtdm/caqtdm
Cc: Mezger Anton Christian (PSI); Comment
Subject: Re: [caqtdm/caqtdm] caSlider improvements (#14)


Dear Anton,

I have push some more bug fixes related to the slider.

  1.  If the sliders step was greater than distance between the slider and the limit it would allow a user to go outside the limits.
  2.  If the slider had a direction of Down or Left and one clicked or held the slider would move in the opposite direction of the click.
  3.  If the slider had a direction of Down or Left the Low limit and High limit was swapped allowing those sliders to go outside the limits.

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-215469365>

AntonMezger commented:
Hi,

I saw that you modified in caqtdm_lib.cpp at the lines 3261 through 3267 the test for the direction out. When I do this after having changed the slider as suggested then the direction of the slider goes in the opposite direction.

Something wrong with your git update ?

Best regards

Anton

From: Josh Dassinger [mailto:notifications@github.com]
Sent: Donnerstag, 28. April 2016 17:36
To: caqtdm/caqtdm
Cc: Mezger Anton Christian (PSI); Comment
Subject: Re: [caqtdm/caqtdm] caSlider improvements (#14)


Dear Anton,

I have push some more bug fixes related to the slider.

  1.  If the sliders step was greater than distance between the slider and the limit it would all user to go outside the limits.
  2.  If the slider had a direction of Down or Left and one clicked or held the slider would move in the opposite direction of the click.
  3.  If the slider had a direction of Down or Left the Low limit and High limit was swapped allowing those sliders to go outside the limits.

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-215469365>

AntonMezger commented:
Perhaps you could send me the changed files (caslider.cpp, caqtdm_lib.cpp, limitsdialog.cpp) directly, while there are some differences due to different versions I think.

Thanks in advance

Best regards

Anton

From: Josh Dassinger [mailto:notifications@github.com]
Sent: Donnerstag, 28. April 2016 17:36
To: caqtdm/caqtdm
Cc: Mezger Anton Christian (PSI); Comment
Subject: Re: [caqtdm/caqtdm] caSlider improvements (#14)


Dear Anton,

I have push some more bug fixes related to the slider.

  1.  If the sliders step was greater than distance between the slider and the limit it would allow a user to go outside the limits.
  2.  If the slider had a direction of Down or Left and one clicked or held the slider would move in the opposite direction of the click.
  3.  If the slider had a direction of Down or Left the Low limit and High limit was swapped allowing those sliders to go outside the limits.

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-215469365>

dassjosh commented:
Anton,

I modified those lines because they were swapping the low limit and high limit when the direction was Left or Down when it shouldn't be doing that. In caSlider i changed how the direction value is calculated. So when the slider is set to Left and you click left of the slider bar it should increment instead of decrementing.

Josh

AntonMezger commented:
Dear Josh,

Perhaps I missed something, so that the best thing is you send me directly the mentioned files.

Best regards

Anton

From: Josh Dassinger [mailto:notifications@github.com]
Sent: Donnerstag, 28. April 2016 18:28
To: caqtdm/caqtdm
Cc: Mezger Anton Christian (PSI); Comment
Subject: Re: [caqtdm/caqtdm] caSlider improvements (#14)


Anton,

I modified those lines because they were swapping the low limit and high limit when the direction was Left or Down when it shouldn't be doing that. In caSlider i changed how the direction value is calculated. So when the slider is set to Left and you click left of the slider bar it should increment instead of decrementing.

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-215485605>

dassjosh commented:
Anton,

I have attached the files

[source_files.zip](https://github.com/caqtdm/caqtdm/files/240970/source_files.zip)


Josh

AntonMezger commented:
Hi Josh,

I found the error, I just inverted the instructions in caqtdm_lib. Thank you for your quick bothering. I saw that you left the instructions about the precision for caslider after the tests. These were already done inside the tests. Just take a look at that.

Best regards

Anton

From: Josh Dassinger [mailto:notifications@github.com]
Sent: Donnerstag, 28. April 2016 18:31
To: caqtdm/caqtdm
Cc: Mezger Anton Christian (PSI); Comment
Subject: Re: [caqtdm/caqtdm] caSlider improvements (#14)


Anton,

I have attached the files

source_files.zip<https://github.com/caqtdm/caqtdm/files/240970/source_files.zip>

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-215486706>

AntonMezger commented:
Dear Josh

I pushed the updated files to git.

Best regards

Anton

From: Josh Dassinger [mailto:notifications@github.com]
Sent: Donnerstag, 28. April 2016 18:31
To: caqtdm/caqtdm
Cc: Mezger Anton Christian (PSI); Comment
Subject: Re: [caqtdm/caqtdm] caSlider improvements (#14)


Anton,

I have attached the files

source_files.zip<https://github.com/caqtdm/caqtdm/files/240970/source_files.zip>

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-215486706>

dassjosh commented:
Dear Anton,

I have updated the slider to step correctly when using the mouse wheel. I also cleaned up the MousePressedEvent and MoveSlider methods.

Josh

AntonMezger commented:
Dear Josh,

Please send me the caslider sources then. I will integrate the changes after testing

Best regards

Anton

From: Josh Dassinger [mailto:notifications@github.com]
Sent: Mittwoch, 18. Mai 2016 16:39
To: caqtdm/caqtdm
Cc: Mezger Anton Christian (PSI); Comment
Subject: Re: [caqtdm/caqtdm] caSlider improvements (#14)


Dear Anton,

I have updated the slider to step correctly when using the mouse wheel. I also cleaned up the MousePressedEvent and MoveSlider methods.

Josh

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/caqtdm/caqtdm/pull/14#issuecomment-220046878>

dassjosh commented:
Here are the source files:
[SourceFiles.zip](https://github.com/caqtdm/caqtdm/files/270754/SourceFiles.zip)
